packages/flutter/test/rendering/paragraph_test.dart

```diff
@@ -125,6 +141,48 @@ void main() {
     expect(boxes.any((ui.TextBox box) => box.right == 100 && box.top == 10), isTrue);
   }, skip: isBrowser); // https://github.com/flutter/flutter/issues/61016
 
+  test('getBoxesForSelection test with boxHeightStyle and boxWidthStyle set to max', () {
+    final RenderParagraph paragraph = RenderParagraph(
+      const TextSpan(
+        text: 'First ',
+        style: TextStyle(fontFamily: 'Ahem', fontSize: 10.0),
+        children: <InlineSpan>[
+          TextSpan(text: 'smallsecond ', style: TextStyle(fontSize: 8.0)),
+          TextSpan(text: 'third fourth fifth'),
+        ],
+      ),
+      textDirection: TextDirection.ltr,
+    );
+    // Do layout with width chosen so that this splits as
+    // First smallsecond |
+    // third fourth |
+    // fifth|
+    // The corresponding line widths come out to be:
+    // 1st line: 156px wide: 6 chars * 10px plus 12 chars * 8px.
+    // 2nd line: 130px wide: 13 chars * 10px.
+    // 3rd line: 50px wide.
+    layout(paragraph, constraints: const BoxConstraints(maxWidth: 160.0));
+
+    final List<ui.TextBox> boxes = paragraph.getBoxesForSelection(
+      const TextSelection(baseOffset: 0, extentOffset: 36),
+      boxHeightStyle: ui.BoxHeightStyle.max,
```

Comment:
Thank you for these tests! If you have the time, it would be great to also add a version of this test with the values at default tight, just to ensure the baseline values are different than when the property is set to max.

Reply:
Done! I created a separate test that does this with the default parameters, but I ended up changing the font-size to 5 instead of 8 because of an issue with floating-point rounding when using 8. Specifically, if you run the old test but with tight boxes, then the second box should have top 1.6 and bottom 9.6, but (at least on my machine) they come out as 1.5999999046325684 etc. instead. With font-size 5 this doesn't seem to happen. This makes the `expect(boxes[1], ...)` check more awkward. Let me know if you'd prefer to keep the font-size 8 version, with some `moreOrLessEquals` tests instead.

Reply:
Changing to 5 is fine! As long as it is tested :)